src/sql/workbench/services/query/common/queryModelService.ts

```diff
@@ -35,23 +35,23 @@ export interface QueryEvent {
 export class QueryInfo {
 	public queryRunner?: EditQueryRunner;
 	public dataService?: DataService;
-	public queryEventQueue?: QueryEvent[];
-	public range?: Array<IRange>;
+	public queryEventQueue: QueryEvent[] = [];
+	public range: Array<IRange> = [];
 	public selectionSnippet?: string;
 
 	// Notes if the angular components have obtained the DataService. If not, all messages sent
 	// via the data service will be lost.
-	public dataServiceReady?: boolean;
+	public dataServiceReady?: boolean = false;
 
-	constructor() {
-		this.dataServiceReady = false;
-		this.queryEventQueue = [];
-		this.range = [];
-	}
+	constructor() { }
 
 	public set uri(newUri: string) {
-		this.queryRunner.uri = newUri;
-		this.dataService.uri = newUri;
+		if (this.queryRunner) {
```

Comment:
While I was here fixed these as well since we're not guaranteed to have them exist when this is called